plugins/clnrest-plugin/src/certs.rs

```diff
@@ -0,0 +1,76 @@
+use anyhow::Error;
+use rcgen::{CertificateParams, DistinguishedName, KeyPair};
+use std::fs;
+use std::net::IpAddr;
+use std::path::PathBuf;
+
+pub fn generate_certificates(certs_path: &PathBuf, rest_host: &str) -> Result<(), Error> {
+    // Generate the CA certificate
+    let mut ca_params = CertificateParams::new(vec![
+        "cln Root REST CA".to_string(),
+        "cln".to_string(),
+        "localhost".to_string(),
+    ])?;
+    ca_params.is_ca = rcgen::IsCa::Ca(rcgen::BasicConstraints::Unconstrained);
+    let ca_key = KeyPair::generate()?;
+    let ca_cert = ca_params.self_signed(&ca_key)?;
+
+    fs::create_dir_all(certs_path)?;
```

Comment:
The `?` will propagate the error and in this case right to the main function where there is another `?` after `generate_certificates` which will stop the plugin with the error message from `fs::create_dir_all`. When does this error:


"This function will return an error in the following situations, but is not limited to just these cases:

If any directory in the path specified by path does not already exist and it could not be created otherwise. The specific error conditions for when a directory is being created (after it is determined to not exist) are outlined by [fs::create_dir].
Notable exception is made for situations where any of the directories specified in the path could not be created as it was being created concurrently. Such cases are considered to be successful. That is, calling create_dir_all concurrently from multiple threads or processes is guaranteed not to fail due to a race condition with itself."